cli/command/stack/cmd.go

```diff
@@ -15,19 +18,24 @@ func NewStackCommand(dockerCli command.Cli) *cobra.Command {
 		RunE:        command.ShowHelp(dockerCli.Err()),
 		Annotations: map[string]string{"version": "1.25"},
 	}
-	cmd.AddCommand(
-		newDeployCommand(dockerCli),
-		newListCommand(dockerCli),
-		newRemoveCommand(dockerCli),
-		newServicesCommand(dockerCli),
-		newPsCommand(dockerCli),
-	)
+	switch orchestrator.GetOrchestrator(dockerCli) {
+	case orchestrator.Kubernetes:
+		kubernetes.AddStackCommands(cmd, dockerCli)
+	case orchestrator.Swarm:
+		swarm.AddStackCommands(cmd, dockerCli)
+	}
```

Comment:
I don't see how this can work if we want to be able to set the orchestrator from the config or a flag, and I think we should support both.

I think this branching needs to be done as part of the `Run()` of the commands.

The same commands exits for both orchestrators, so the implementation of the `Run()` can be switch and call the `Run()` for the correct orchestrator.